From 009e92adb9b4dbb347867e53d5831607e9cecbcb Mon Sep 17 00:00:00 2001
From: Peter O'Connor <peter@solus-project.com>
Date: Sat, 4 Aug 2018 15:08:29 +1000
Subject: [PATCH] cmd/snap-confine: allow confinement of AVX specific library
 paths

Since glibc 2.26 library lookups will now look first for specific AVX2 and
AVX512 library versions on compatible hardware. This patch adds the directories
to snap-confine, allowing snap to support distributions that ship these
libraries.

Before:
/usr/lib/snapd/snap-confine: error while loading shared libraries: libc.so.6:
failed to map segment from shared object

After:
Application loads as expected
---
 cmd/snap-confine/snap-confine.apparmor.in | 38 +++++++++++------------
 1 file changed, 19 insertions(+), 19 deletions(-)

diff --git a/cmd/snap-confine/snap-confine.apparmor.in b/cmd/snap-confine/snap-confine.apparmor.in
index 75d7ce073b..923c2a3713 100644
--- a/cmd/snap-confine/snap-confine.apparmor.in
+++ b/cmd/snap-confine/snap-confine.apparmor.in
@@ -14,27 +14,27 @@
     # any abstractions
     /etc/ld.so.cache r,
     /etc/ld.so.preload r,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}ld-*.so mrix,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}ld-*.so mrix,
     # libc, you are funny
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libc{,-[0-9]*}.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libpthread{,-[0-9]*}.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libreadline{,-[0-9]*}.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}librt{,-[0-9]*}.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libgcc_s.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libncursesw{,-[0-9]*}.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libresolv{,-[0-9]*}.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libselinux.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libpcre.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libc{,-[0-9]*}.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libpthread{,-[0-9]*}.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libreadline{,-[0-9]*}.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}librt{,-[0-9]*}.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libgcc_s.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libncursesw{,-[0-9]*}.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libresolv{,-[0-9]*}.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libselinux.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libpcre.so* mr,
     # normal libs in order
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libapparmor.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libcgmanager.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libdl{,-[0-9]*}.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libnih.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libnih-dbus.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libdbus-1.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libudev.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libseccomp.so* mr,
-    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libcap.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libapparmor.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libcgmanager.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libdl{,-[0-9]*}.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libnih.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libnih-dbus.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libdbus-1.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libudev.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libseccomp.so* mr,
+    /{,usr/}lib{,32,64,x32}/{,haswell/{,avx512_1/},@{multiarch}/}libcap.so* mr,

     @LIBEXECDIR@/snap-confine mr,

